lib/PuppeteerSharp.Tests/WaitTaskTests/FrameWaitForFunctionTests.cs

```diff
@@ -216,5 +232,59 @@ public async Task ShouldSurviveNavigations()
             await Page.EvaluateFunctionAsync("() => window.__done = true");
             await watchdog;
         }
+
+        private TaskCompletionSource<bool> ListenForStartPolling()
+        {
+            TaskCompletionSource<bool> startedPolling = new TaskCompletionSource<bool>();
+
+            // Wait for function will release the execution faster than in node.
+            // We intercept the poller.start() call to prevent tests from continuing before the polling has started.
+            _connectionTransportInterceptor.MessageSent += (_, message) =>
+            {
+                if (message.Contains("poller => poller.start()"))
+                {
+                    startedPolling.SetResult(true);
+                }
+            };
+
+            return startedPolling;
+        }
+
+        private sealed class ConnectionTransportInterceptor : IConnectionTransport
```

Comment:
Let's move this to its own file and make it public. This is a great idea.

Reply:
Should I create a new folder, or leave it in WaitTaskTests?

Reply:
Let's move it to the root folder by now. There are many classes there that could be moved to an `Utils` directory.